graphs/search/depthFirstSearch.go

```diff
@@ -0,0 +1,36 @@
+// Package search Graph search algorithms
+// reference: https://en.wikipedia.org/wiki/Tree_traversal
+package search
+
+import (
+	"github.com/TheAlgorithms/Go/datastructures/stack"
+)
+
+/*DepthFirstSearch is an algorithm for traversing or searching graph data structures.
+	The algorithm starts at the start node and explores as far as possible along each branch before backtracking.
+	Worst-case performance			O(|V|+|E|)}O(|V| + |E|)
```

Comment:
The comment is good, but could you add what you mean by `V` and `E` - not many students would know that this refers to vertices and edges in a graph

Reply:
I think there is no need for more explanation. for more info about the algorithm, big-O, and other extra data there is a reference in the last line of the comment. prolonging the comments may have a negative effect.
this is my opinion but if you think more comment is better I will add another commit.

Reply:
The comments are useful for making the documentation for this repository self-contained. Just visit [godoc.org](https://pkg.go.dev/github.com/TheAlgorithms/Go), you can see that quite a lot of code is not very well documented - precisely because the comments are missing. I personally think it would be very good for the documentation to be as self-contained as possible.